src/d3a/d3a_core/sim_results/market_price_energy_day.py

```diff
@@ -0,0 +1,111 @@
+from collections import OrderedDict
+from statistics import mean
+
+from d3a.d3a_core.util import round_floats_for_ui
+
+
+class MarketPriceEnergyDay:
+    def __init__(self, should_export_plots):
+        self._price_energy_day = {}
+        self.csv_output = {}
+        self.redis_output = {}
+        self.should_export_plots = should_export_plots
+
+    @classmethod
+    def gather_trade_rates(cls, area_dict, core_stats, current_market_slot, price_lists,
+                           fee_lists):
+        if area_dict['children'] == []:
+            return price_lists, fee_lists
+
+        cls.gather_rates_one_market(area_dict, core_stats, current_market_slot, price_lists,
+                                    fee_lists)
+
+        for child in area_dict['children']:
+            price_lists, fee_lists = cls.gather_trade_rates(child, core_stats, current_market_slot,
+                                                            price_lists, fee_lists)
+
+        return price_lists, fee_lists
+
+    @classmethod
+    def gather_rates_one_market(cls, area_dict, core_stats, current_market_slot, price_lists,
+                                fee_lists):
+        if area_dict['uuid'] not in price_lists:
+            price_lists[area_dict['uuid']] = OrderedDict()
+        if current_market_slot not in price_lists[area_dict['uuid']].keys():
+            price_lists[area_dict['uuid']][current_market_slot] = []
+        trade_rates = [
+            # Convert from cents to euro
+            t['energy_rate'] / 100.0
+            for t in core_stats.get(area_dict['uuid'], {}).get('trades', [])
+        ]
+        price_lists[area_dict['uuid']][current_market_slot].extend(trade_rates)
+
+        if area_dict['uuid'] not in fee_lists:
+            fee_lists[area_dict['uuid']] = OrderedDict()
+        if current_market_slot not in fee_lists[area_dict['uuid']].keys():
+            fee_lists[area_dict['uuid']][current_market_slot] = []
+        fee_rates = [
+            # Convert from cents to euro
+            t['fee_price'] / 100.0 / t['energy']
+            for t in core_stats.get(area_dict['uuid'], {}).get('trades', [])
```

Comment:
A potential improvement with regard of computational effort would be to use one for loop only to populate `fee_rates` and `trade_rates`. What do you think?

Reply:
Outdated, as fee being set at the market only needs to be plotted.